### ISSUE TITLE
dmypy does not detect the --no-color option

### ISSUE THREAD

**Issue by peendebak** [python/mypy#12289]
**Bug Report**

The mypy daemon does not parse the `--no-color` correctly on windows.  The output of `dmypy` is not properly displayed in a windows console.

(there are two issues here: `dmypy` ignoring `--no-color` and the problems in displaying colors)

**To Reproduce**

1. Run `dmypy` from a windows termimal

**Expected Behavior**

The output of `dmypy` should render colors correctly. Not having any colored output at all is also acceptable.
The mypy daemon does not parse color output correctly on windows. This could be resolved by using the `--no-color` option, but this is not used by `dmpypy`.
 
**Actual Behavior**

See the screenshot below: the invocations of `mypy` work fine, but the output for `dmpypy` hs hard to interpret.

![image](https://user-images.githubusercontent.com/17724047/156751153-e31aa793-d69c-47c5-b7f5-c35269026994.png)

Related: #8164

**Your Environment**

- Mypy version used: 0.931
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.9
- Operating system and version: Windows 10

**Comment by Andrioden** [comment]
Seeing the same problem, Windows 10 for powershell, git bash and command promtp

**Comment by DStauffman** [comment]
Any progress on this?  mypy takes about 3 mins to run on my code base, where dmypy only takes a couple seconds.  So mypy is unusable, but the dmypy readout is incredibly hard to parse and really annoying.

**Comment by hauntsaninja** [comment]
Want to make a PR?
